refactor(cfg): use IndexVec for storing basic blocks

DonIsaac opened the pull request:
Use an `IndexVec` when storing basic blocks. This makes the link between nodes in `.graph` and elements of `.basic_blocks` more clear. I had to rename `BasicBlockId` to `BlockNodeId` to avoid a name collision. I wasn't sure what else to name the `Idx` type for the basic blocks vec.

DonIsaac commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6323?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6323** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6323?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#6322** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6322?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6321** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6321?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6320** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6320?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6319** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6319?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
3 things:

Why is `BlockNodeId` a better name than `BasicBlockId`? I don't know much about CFGs, but isn't "basic block" a term that's usual with CFGs?

If you *are* making that change, it'd be much easier to review a PR like this if the unimportant change - renaming the type - was a separate PR, so that the substantive change stands out. It's quite hard to wade through the diff in a PR like this, and figure out what's going on.

It's really helpful if you can put *some* description on PRs. It's doesn't need to be an essay, but just something short indicating the purpose of/motivation for the change, and any salient points about the implementation. The code usually is self-explanatory concerning the "what", but it often doesn't give you the "why".

DonIsaac commented:
@overlookmotel thanks for the feedback. I've added a description, and it should answer your question.

DonIsaac committed:
refactor(cfg): use IndexVec for storing basic blocks (#6323)

Use an `IndexVec` when storing basic blocks. This makes the link between nodes in `.graph` and elements of `.basic_blocks` more clear. I had to rename `BasicBlockId` to `BlockNodeId` to avoid a name collision. I wasn't sure what else to name the `Idx` type for the basic blocks vec.
